libs/methodologies/bold/rule-processors/mass-id/src/weighing/weighing.helpers.ts

```diff
@@ -111,207 +165,258 @@ export const getWeighingEvents = (document: Document): DocumentEvent[] =>
     [],
   );
 
-export const validateContainerCapacityAttribute = (
-  containerCapacityAttribute: MethodologyDocumentEventAttribute | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (!hasPositiveValue(containerCapacityAttribute)) {
-    rejectedMessages.push(MISSING_RESULT_COMMENTS.CONTAINER_CAPACITY);
-  }
-
-  if (!hasKilogramFormat(containerCapacityAttribute)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.CONTAINER_CAPACITY_FORMAT);
-  }
-
-  return rejectedMessages;
-};
-
-export const validateContainerQuantity = (
-  containerQuantity: MethodologyDocumentEventAttributeValue | undefined,
-  vehicleType: MethodologyDocumentEventAttributeValue | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (vehicleType === TRUCK && isNonZeroPositiveInt(containerQuantity)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.CONTAINER_QUANTITY);
-  }
-
-  if (!isNonZeroPositiveInt(containerQuantity)) {
-    rejectedMessages.push(MISSING_RESULT_COMMENTS.CONTAINER_QUANTITY);
-  }
-
-  return rejectedMessages;
-};
-
-export const validateGrossWeightAttribute = (
-  grossWeight: MethodologyDocumentEventAttribute | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (!hasPositiveValue(grossWeight)) {
-    rejectedMessages.push(
-      MISSING_RESULT_COMMENTS.GROSS_WEIGHT(grossWeight?.value),
-    );
-  }
-
-  if (!hasKilogramFormat(grossWeight)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.GROSS_WEIGHT_FORMAT);
-  }
-
-  return rejectedMessages;
-};
-
-export const validateMassNetWeightAttribute = (
-  massNetWeightAttribute: MethodologyDocumentEventAttribute | undefined,
-): ValidationResult => {
-  const rejectedMessages: string[] = [];
-
-  if (!hasPositiveValue(massNetWeightAttribute)) {
-    rejectedMessages.push(
-      MISSING_RESULT_COMMENTS.MASS_NET_WEIGHT(massNetWeightAttribute?.value),
-    );
-  }
-
-  if (!hasKilogramFormat(massNetWeightAttribute)) {
-    rejectedMessages.push(INVALID_RESULT_COMMENTS.MASS_NET_WEIGHT_FORMAT);
-  }
+type Validator = (values: WeighingValues, isTwoStep?: boolean) => string[];
```

Comment:
I think the result is clearer if the errors are returned in a object with errors key
```suggestion
type Validator = (values: WeighingValues, isTwoStep?: boolean) => { errors: string[] };
```

Reply:
https://github.com/carrot-foundation/methodology-rules/pull/200/commits/263989983a281062e10c70f03b588460d950048e